### ISSUE TITLE
Could not parse template name "@VendorBundle/template.html.twig". Expected format is "scope:name.format"

### ISSUE THREAD

**Issue by fritzmg** [netzmacht/contao-toolkit#26]
In one of our extensions we are rendering a twig template using the `render` method of the `Symfony\Bundle\FrameworkBundle\Controller\AbstractController`. However, when `netzmacht/contao-toolkit` is installed, the following error happens:

```
RuntimeException:
Could not parse template name "@FoobarBundle/template.html.twig". Expected format is "scope:name.format"

  at vendor/netzmacht/contao-toolkit/src/View/Template/ToolkitTemplateNameParser.php:42
  at Netzmacht\Contao\Toolkit\View\Template\ToolkitTemplateNameParser->parse('@FoobarBundle/template.html.twig')
     (vendor/netzmacht/contao-toolkit/src/View/Template/ToolkitTemplateEngine.php:140)
  at Netzmacht\Contao\Toolkit\View\Template\ToolkitTemplateEngine->getTemplateReference('@FoobarBundle/template.html.twig')
     (vendor/netzmacht/contao-toolkit/src/View/Template/ToolkitTemplateEngine.php:64)
  at Netzmacht\Contao\Toolkit\View\Template\ToolkitTemplateEngine->render('@FoobarBundle/template.html.twig', array(…))
     (vendor/symfony/framework-bundle/Controller/ControllerTrait.php:233)
  at Symfony\Bundle\FrameworkBundle\Controller\AbstractController->render('@FoobarBundle/template.html.twig', array(…))
     (vendor/foobar/contao-foobar/src/Controller/BackendController.php:59)
  at InspiredMinds\FoobarBundle\Controller\BackendController->templateAction(object(Request))
     (vendor/symfony/http-kernel/HttpKernel.php:158)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/http-kernel/HttpKernel.php:80)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/http-kernel/Kernel.php:201)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (web/index.php:31)
```
`@VendorBundle/path/to/template.html.twig` is the correct way to reference bundle templates. See https://symfony.com/doc/4.4/templates.html#bundle-templates

**Comment by fritzmg** [comment]
As a work around I am not using the `AbstractController` anymore. Instead I am using the `@twig` service directly.

**Comment by dmolineus** [comment]
The toolkit (still) depends on the templating engine. Therefor it activates the `toolkit` template engine. Because of the deprecation of the template engine (and because Contao does not activate the `twig` engine anymore, the toolkit template engine is used all the time in the `Symfony\Bundle\FrameworkBundle\Controller\ControllerTrait::renderView`: https://github.com/symfony/symfony/blob/b60bb6e2d6051f150ea06dd794ae2894e79aacd8/src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php#L210-L214

Imo this is the bug of the symfony controller trait. Assuming that the template engine can render a template is clearly wrong, at leat the `EngineInterface::supports` should be called here. https://github.com/symfony/templating/blob/42fb44375553828eee4fa12bd255579b9ccfc871/EngineInterface.php#L63

As a workaround enable the twig template engine:

```yaml
framework:
  templating:
    engines: 
      - twig
```

**Comment by fritzmg** [comment]
Interesting. So it should probably be:

```php
if ($this->container->has('templating')) {
    @trigger_error('Using the "templating" service is deprecated since version 4.3 and will be removed in 5.0; use Twig instead.', \E_USER_DEPRECATED);

    if ($this->container->get('templating')->supports($view)) {
        return $this->container->get('templating')->render($view, $parameters);
    }
}
```

correct? I'll make some tests and then may be a PR for Symfony 4.4 :)

**Comment by dmolineus** [comment]
Yes, I think so. I'd prefer the second one. The templating engine support will be dropped from the toolkit by the way, but can only happen in the next major version.

**Comment by dmolineus** [comment]
Did you make any progress here?

**Comment by fritzmg** [comment]
Never got around to it back then, but done now: https://github.com/symfony/symfony/pull/40373 :)

**Comment by dmolineus** [comment]
Thanks, otherwise I would have created one :-)

**Comment by fritzmg** [comment]
PR has been merged, should be fixed in Symfony `4.4.21` and up.
